src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -3837,6 +3837,213 @@ static void Main()
             Assert.True(type.IsNativeIntegerType);
         }
 
+        [Theory, CombinatorialData]
+        [WorkItem(60337, "https://github.com/dotnet/roslyn/issues/60337")]
+        public void NameOfNintSpeculation_01([CombinatorialValues("always", "never")] string runNullableAnalysis)
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        var x = nameof(nint);
+    }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9.WithFeature("run-nullable-analysis", runNullableAnalysis));
+            comp.VerifyDiagnostics(
+                // (6,24): error CS0103: The name 'nint' does not exist in the current context
+                //         var x = nameof(nint);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "nint").WithArguments("nint").WithLocation(6, 24)
+            );
+            var tree = comp.SyntaxTrees.Single();
+            var root = tree.GetRoot();
+            var existingNint = root.DescendantNodes().OfType<ArgumentSyntax>().Single().Expression;
+            var existingNameOf = root.DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+            var model = comp.GetSemanticModel(tree);
+            var nameofToSpeculate = (InvocationExpressionSyntax)SyntaxFactory.ParseExpression("nameof(nint)");
```

Comment:
>nameof [](http://example.com/codeflow?start=95&length=6)

The presence of `nameof` here makes no difference because we are not speculating on the resulting ```InvocationExpressionSyntax```. This confuses the reader and because of that potentially sets wrong expectations.